src/main/java/org/wso2/carbon/connector/util/AbstractAzureMediator.java

```diff
@@ -0,0 +1,232 @@
+                                                             String ifModifiedSince, String ifUnmodifiedSince) {
+
+        return new DataLakeRequestConditions()
+                .setLeaseId(leaseId)
+                .setIfMatch(ifMatch)
+                .setIfModifiedSince(ifModifiedSince != null ? OffsetDateTime.parse(ifModifiedSince) : null)
+                .setIfNoneMatch(ifNoneMatch)
+                .setIfUnmodifiedSince(ifUnmodifiedSince != null ? OffsetDateTime.parse(ifUnmodifiedSince) : null);
+    }
+
+    protected void handleConnectorResponse(MessageContext messageContext, String responseVariable,
+                                           Boolean overwriteBody, Object payload,
+                                           Map<String, Object> headers, Map<String, Object> attributes) {
+
+        ConnectorResponse response = new DefaultConnectorResponse();
+        if (payload == null) {
+            // Empty json object
+            payload = Map.of();
+        }
+        if (headers == null) {
+            headers = Map.of();
+        }
+        if (attributes == null) {
+            attributes = Map.of();
+        }
+
+        if (overwriteBody != null && overwriteBody) {
+
+            org.apache.axis2.context.MessageContext axisMsgCtx =
+                    ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+            try {
+                JsonUtil.getNewJsonPayload(axisMsgCtx, payload.toString(), true, true);
+
+            } catch (AxisFault e) {
+                handleException("Error setting response payload", e, messageContext);
+            }
+
+            axisMsgCtx.setProperty(org.apache.axis2.Constants.Configuration.MESSAGE_TYPE,
+                    AzureConstants.JSON_CONTENT_TYPE);
+            axisMsgCtx.setProperty(org.apache.axis2.Constants.Configuration.CONTENT_TYPE,
+                    AzureConstants.JSON_CONTENT_TYPE);
+
+        } else {
+            response.setPayload(payload);
+        }
+        response.setHeaders(headers);
+        response.setAttributes(attributes);
+        messageContext.setVariable(responseVariable, response);
+    }
+
+    public void handleConnectorException(Error code, MessageContext mc, Throwable e) {
+
+        this.log.error(code.getErrorMessage(), e);
+
+        mc.setProperty("ERROR_CODE", code.getErrorCode());
+        mc.setProperty("ERROR_MESSAGE", code.getErrorMessage());
+        throw new SynapseException(code.getErrorMessage(), e);
+    }
+
+    public void handleConnectorException(Error code, MessageContext mc) {
+
+        this.log.error(code.getErrorMessage());
+        mc.setProperty("ERROR_CODE", code.getErrorCode());
+        mc.setProperty("ERROR_MESSAGE", code.getErrorMessage());
+        throw new SynapseException(code.getErrorMessage());
+    }
+
+    /**
+     * Get LeaseAction object.
+     *
+     * @param leaseAction Lease action.
+     * @return LeaseAction object.
+     */
+    public LeaseAction getLeaseAction(String leaseAction) {
+
```

Comment:
Shall we check whether the leaseAction is null and return accordingly to avoid unnecessary processing of the switch block?